stores/metadata.go

```diff
@@ -213,6 +213,14 @@ func (s *SQLStore) ContractRoots(ctx context.Context, id types.FileContractID) (
 	return
 }
 
+func (s *SQLStore) ContractRootsDiff(ctx context.Context, fcid types.FileContractID, roots []types.Hash256, offset uint64) (indices []uint64, err error) {
```

Comment:
Imo it's confusing to have an `offset` parameter that is passed down to the db code. It imposes responsibility on the store that it shouldn't have. It's fine for the returned indices to correspond to the indices of the input slice and then have the caller who processes these batches to transform those indices back into the "global" scope.